lib/galaxy/datatypes/isa.py

```diff
@@ -448,26 +445,36 @@ def display_data(self, trans, dataset, preview=False, filename=None, to_ext=None
                 html += '<p>Submitted the %s</p>' % study.submission_date
                 html += '<p>Released on %s</p>' % study.public_release_date
 
+                html += '<p>Experimental factors used: %s</p>' % ', '.join([x.name for x in study.factors])
+
                 # Loop on all assays of this study
                 for assay in study.assays:
                     html += '<h3>Assay %s</h3>' % assay.filename
-                    html += '<p>Measurement type: %s</p>' % assay.measurement_type.term # OntologyAnnotation
-                    html += '<p>Technology type: %s</p>' % assay.technology_type.term # OntologyAnnotation
+                    html += '<p>Measurement type: %s</p>' % assay.measurement_type.term  # OntologyAnnotation
+                    html += '<p>Technology type: %s</p>' % assay.technology_type.term    # OntologyAnnotation
                     html += '<p>Technology platform: %s</p>' % assay.technology_platform
                     if assay.data_files is not None:
-                        html += '<p>Data files:</p>'
-                        html += '<ul>'
-                        for data_file in assay.data_files:
-                            html += '<li>' + str(data_file.id) + ' - ' + str(data_file.filename) + ' - ' + str(data_file.label) + '</li>'
-                        html += '</ul>'
-
-            html += '</body></html>'
+                        file_types = set(x.label for x in assay.data_files)
+                        data_files = {}
+                        for file_type in file_types:
+                            data_files[file_type] = sorted([x.filename for x in
+                                                     assay.data_files if
+                                                     x.label == file_type])
+                        for label, filenames in data_files.items():
```

Comment:
use an iterator (`iteritems()`) instead of creating a new list with `items()`

Reply:
Sure I can update this. It's worth noting that Python 3 `items()` returns an iterator and there is no `iteritems()`, so whenever Galaxy moves to py3 this will be changed back again...

Reply:
Indeed, this is an issue.  Given what you reasonably expect this code to see as ISA datasets, it might not be worth changing.  Alternatively, there are some idioms that work well with both Python 2 and 3.  For instance, there's one using `six`, which Galaxy already uses:

```
from six import iteritems
for (key, value) in iteritems(heights):
    ...
```

I pulled that tip from this page :-)  http://python-future.org/compatible_idioms.html